File: edgequake/sdks/java/src/main/java/io/edgequake/sdk/EdgeQuakeException.java
package io.edgequake.sdk;

/**
 * Exception thrown when the EdgeQuake API returns a non-2xx status.
 */
public class EdgeQuakeException extends RuntimeException {

    private final int statusCode;
    private final String responseBody;

    public EdgeQuakeException(int statusCode, String responseBody) {
        super("EdgeQuake API error " + statusCode + ": " + responseBody);
        this.statusCode = statusCode;
        this.responseBody = responseBody;
    }

    public EdgeQuakeException(String message, Throwable cause) {
        super(message, cause);
        this.statusCode = 0;
        this.responseBody = "";
    }

    public int statusCode() { return statusCode; }
    public String responseBody() { return responseBody; }
}

File: edgequake/sdks/kotlin/src/main/kotlin/io/edgequake/sdk/EdgeQuakeException.kt
package io.edgequake.sdk

/** WHY: Typed exception with status code for callers to handle HTTP errors. */
class EdgeQuakeException(
    message: String,
    val statusCode: Int = 0,
    val responseBody: String? = null,
    cause: Throwable? = null
) : RuntimeException(message, cause)
